### ISSUE TITLE
Generic handling to ignore errors on startup

### ISSUE THREAD

**Issue by powersj** [influxdata/telegraf#14803]
### Use Case

In numerous feature requests, users are asking for a way to ignore errors on start up. It would be nice to have a high-level generic method to add this to input plugins.

### Expected behavior

Agent level setting to set this behavior for all input plugins.

### Actual behavior

Currently users will see Telegraf throw an error during start up if an input plugin cannot start. There are a couple exceptions to this where a configuration option is present to avoid these:

```s
  ## Behavior when we fail to connect to the endpoint on initialization. Valid options are:
  ##     "error": throw an error and exits Telegraf
  ##     "ignore": ignore this plugin if errors are encountered
  # connect_fail_behavior = "error"
```

**Comment by srebhan** [comment]
We agreed on
```toml
  ## Behavior when starting the plugin fail e.g. due to connectivity issues. Valid options are:
  ##     "error": exit Telegraf with an error (current behavior)
  ##     "ignore": remove this plugin from further processing if errors are encountered
  ##     "retry": retry starting up the plugin in each gather or flush cycle
  # startup_error_behavior = "error"

**Comment by miken32** [comment]
Is it safe to assume the linked PR will apply to input plugins as well as output? I'm thinking specifically of #12959 here, where we're in the absurd situation of losing reporting for dozens of hosts if one of them goes offline.

**Comment by srebhan** [comment]
@miken32 the target is to provide the functionality for both inputs and outputs. The first PR will tackle the output side (more specifically the framework itself plus one example plugin) but additional PRs will eventually also implement startup-retry _framework_ for input plugins.

Regarding the mentioned issue #12959: This will not be solved by the startup-retry framework alone as it seems like the auth-information is gathered only once in `Init()` which is wrong as this has to be in `Start()` to benefit from the framework. In any case, your hosts will still be missing as long as the authentication server is unreachable!

**Comment by srebhan** [comment]
Reopening as we still need to do the input side...
